terra_notebook_utils/blobstore/azure_blob_store.py

```diff
@@ -0,0 +1,60 @@
+from terra_notebook_utils import blobstore
+from azure.storage.blob import BlobServiceClient
+import os
+from azure.identity import DefaultAzureCredential
+
+class AzureBlobStore(blobstore.BlobStore):
+    schema = "https://"
```

Comment:
This schema is unfortunately awkward for TNU. Currently, the CLI command to copy a drs file into a Google bucket is
```
tnu drs copy drs://foo gs://my-bucket/my-key
```

Note the `gs://` schema. The analogous command for an azure container looks awkward due to the `https://` schema
```
tnu drs copy drs://foo https://some-weird-azure-url
```

A further complication is URL detection. TNU uses the schema to understand the storage provider for destination URLs. Logic for detecting Azure destinations will need to be added [here](https://github.com/DataBiosphere/terra-notebook-utils/blob/f326afb538da730f0f54ad015c125323a64065f1/terra_notebook_utils/blobstore/copy_client.py#L91-L101).